src/tests/functional/navigation_tests.ts

```diff
@@ -116,6 +116,22 @@ export class NavigationTests extends TurboDriveTestCase {
     this.assert.equal(await this.pathname, "/src/tests/fixtures/one.html")
     this.assert.equal(await this.visitAction, "restore")
   }
+
+  async "test skip link with hash-only path"() {
+    await this.clickSelector('a[href="#main"]')
+    await this.nextBeat
+
+    this.assert.equal(await this.pathname, "/src/tests/fixtures/navigation.html")
+    this.assert.equal(await this.hash, "#main")
+    this.assert.ok(await this.isScrolledToSelector("#main"))
+
+    await this.remote.pressKeys('\uE004') // TAB
+    const activeElement = await this.remote.getActiveElement()
+    const skippedLink = await this.querySelector("#ignored-link")
+    const firstLinkWithinMain = await this.querySelector("#main a:first-of-type")
```

Comment:
@sstephenson I was hoping that https://github.com/hotwired/turbo/commit/5525c21288edc222de0a05c72538aa7f898abd0c would resolve this issue, but it hasn't.